src/DensityTest.jl

```diff
@@ -0,0 +1,272 @@
+# Density test for manipulation in the running variable. 
+# This work is based on McCrary, J. (2008). Manipulation of the running variable in the regression discontinuity design: A density test. Journal of econometrics, 142(2), 698-714.
+
+"""
+    McCraryTest(bin::Union{Real,Nothing} = nothing, bw::Union{Real,Nothing} = nothing) 
+
+A McCrary (2008) test for manipulation in the running variable for `SharpRD` estimator.
+
+The `bin` object is the width of each bin. The default is 2*sd(runvar)*length(runvar)^(-0.5), following McCrary 2008.
+
+The `bw` object is the bandwidth. The default uses the calculation from McCrary (2008), pg. 705.
+"""
+Base.@kwdef struct McCraryTest 
```

Comment:
Instead of `nothing` maybe replace by `:mccrary_default` or so? An even better alternative would be to create new types similar to [`ImbensKalyanarman`](https://github.com/nignatiadis/RegressionDiscontinuity.jl/blob/74051523aee41b6ab79d21e4e92770f4b2773556/src/imbens_kalyanaraman.jl#L7).

Reply:
I don't totally understand what you want here.